### PR TITLE
Rewrite with pulp-glue

### PR DESCRIPTION
**Pull request by mdellweg**
This is no more than a PoC at this stage, but the ultimate goal is to use the context framework of the pulp-cli in the ansible modules.

### CODE REVIEW

File: plugins/module_utils/pulp_glue.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+            username=dict(required=True, fallback=(env_fallback, ["SQUEEZER_USERNAME"])),
+            password=dict(
+                required=True,
+                no_log=True,
+                fallback=(env_fallback, ["SQUEEZER_PASSWORD"]),
+            ),
+            validate_certs=dict(
+                type="bool",
+                default=True,
+                fallback=(env_fallback, ["SQUEEZER_VALIDATE_CERTS"]),
```
Comment (gerrod3) [inline plugins/module_utils/pulp_glue.py:65]: How does `fallback` work with a `default` value?

File: plugins/module_utils/pulp_glue.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+            supports_check_mode=supports_check_mode,
+            **kwargs,
+        )
+
+        for import_error in import_errors:
+            if import_error[1] is not None:
+                self.fail_json(msg=missing_required_lib(import_error[0]), exception=import_error[1])
+
+        self.pulp_ctx = PulpSqueezerContext(
+            api_root="/pulp/",
```
Comment (gerrod3) [inline plugins/module_utils/pulp_glue.py:86]: Should `api_root` not be configurable?

File: plugins/modules/delete_orphans.py [hunk 3]
```diff
@@ -38,15 +43,43 @@
...
+                item["message"].split(" ")[-1].lower(): item["total"]
+                for item in task["progress_reports"]
+            }
+        else:
+            # Fake it
+            summary = {
+                "artifacts": 0,
+                "content": 0,
+            }
+        module.set_changed()
```
Comment (gerrod3) [inline plugins/modules/delete_orphans.py:82]: Is there a way to have the`set_changed` only trigger for this step if orphan cleanup actually did work? Would that be in-line with ansible's idempotent philosophy?

File: plugins/modules/task.py [hunk 4]
```diff
@@ -58,14 +58,55 @@
...
+                if not self.check_mode:
+                    if self.state == "canceled":
+                        self.context.cancel()
+                        # This should all be part of the cancel call...
+                        try:
+                            entity = self.pulp_ctx.wait_for_task(entity)
+                        except PulpException as e:
+                            if "canceled" in str(e):
+                                self.context.pulp_href = entity["pulp_href"]
+                                entity = self.context.entity
```
Comment (gerrod3) [inline plugins/modules/task.py]: Is this a TODO for pulp-glue?

**Review by gerrod3 — COMMENTED** [review]
I looked over the new modules and they all seem good to me. Left some extra non-blocking questions for understanding. Also, is this only for pulp_file, does that mean there will be as a large a change when you update it for the other plugins?

File: plugins/module_utils/pulp_glue.py (hunk 2 above)
Comment (mdellweg) [reply plugins/module_utils/pulp_glue.py:86]: Yes, but not necessarily for feature parity...

File: plugins/modules/delete_orphans.py (hunk 3 above)
Comment (mdellweg) [reply plugins/modules/delete_orphans.py:82]: I do not really know. This module always triggers a task / issues a POST|PUT|PATCH|DELETE request. So i consider it changing something.
Also when running in check_mode, there is no way to know that nothing would happen.
It is an action module after all not having a state parameter...

File: plugins/modules/task.py (hunk 4 above)
Comment (mdellweg) [reply plugins/modules/task.py]: Not anymore. But it's not yet released...

File: plugins/module_utils/pulp_glue.py (hunk 1 above)
Comment (mdellweg) [reply plugins/module_utils/pulp_glue.py:65]: I can only guess... Probably set value, then env_fallback, then default.

**Pull request merged**